src/compas/datastructures/mesh/offset.py

```diff
@@ -84,7 +84,10 @@ def mesh_thicken(mesh, thickness=1.0, cls=None):
 
     # close boundaries
     n = thickened_mesh.number_of_vertices() / 2
-    for u, v in list(thickened_mesh.edges_on_boundary()):
+
+    edges_on_boundary = [edge for boundary in list(thickened_mesh.edges_on_boundaries()) for edge in boundary]
```

Comment:
this seems wrong. will this not create one face for all boundaries? should it not create a face per boundary?

Reply:
yes this creates a face per edge on the boundary. but if a mesh has more than one closed boundary, it will just create faces for one boundary and ignore the rest.
![image](https://user-images.githubusercontent.com/44199787/96731653-cb8eac00-13b7-11eb-90a5-772ea08c7a43.png)

Reply:
maybe i misunderstand, but it seems to me that you are making one flat list for all edges on all boundaries. and then you add a face for those edges.

which seems not correct...

Reply:
yes that is what I was doing. but isn't what we want? to add a face to each of these edges(blue and red), so the thickened mesh can be closed? the test result looks correct to me. There are no overlapped faces here if that is what you are concerned. While using the previous version the red loop will be missing.
![image](https://user-images.githubusercontent.com/44199787/96859820-42807f00-1462-11eb-8485-f58817f0a6d2.png)

Reply:
ha sorry, that indeed means i misunderstood...